sdk/core/Azure.Core/src/SequenceWriter.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// A buffer writer which writes large sequences of data into smaller shared buffers.
+    /// </summary>
+    internal sealed class SequenceWriter : IBufferWriter<byte>, IDisposable
+    {
+        private struct Buffer
+        {
+            public byte[] Array;
+            public int Written;
+        }
+
+        private Buffer[] _buffers; // this is an array so items can be accessed by ref
+        private int _count;
```

Comment:
I have probably missed some part of conversation, but why use separate field for `_count`?

Reply:
this represents how many buffers are currently used vs how many buffers are available.  We grow the _buffers whenever _count == _buffers.Length